### PR TITLE
View favourites use case

### PR DESCRIPTION
**Pull request by choisar4**
I made some changes to the View Favourites View. It now has buttons so that you can access the Restaurant views for your favourite restaurants. I have also modified the Restaurant view so that the return button always returns to the previous page (either view restaurants view or view favourites view). No conflicts with base branch for now.

### CODE REVIEW

File: src/view/RestaurantView.java
Comment (HarryC493) [inline src/view/RestaurantView.java:1]: Really smart work being done here by making the state record what previous view was. In our very first group meeting, we mentioned this potential problem that could be tricky to solve. But now this problem has been solved. Good job.

**Review by HarryC493 — CHANGES_REQUESTED** [review]
I've pulled this branch and run the program on my local computer. Everything works well. Just make sure to empty or delete the csv files (namely the favourite.csv) so that we will have no merge conflict with csv files in future pull requests.

**Comment by choisar4** [comment]
I've resolved the conflict and fixed a minor issue with the add to favourites button disappearing. I've made the decision to keep the button at all times for now. Perhaps later we can consider removing it when accessing the restaurant view from the favourites list.

**Review by HarryC493 — APPROVED** [review]
Thanks for resolving the merge conflict from the csv files. I'll merge this branch to the main branch. Good job.

**Pull request merged**